### PR TITLE
Fix infinite scroll in process details area

### PR DESCRIPTION
**Pull request by grouzen**
Addresses #54

### CODE REVIEW

File: src/tui/rendering.rs [hunk 1]
```diff
@@ -37,12 +37,13 @@ impl Theme {
...
     theme: Theme,
-    number_of_items: usize,
-    details_scroll_state: ScrollbarState,
-    process_details_scroll: u16,
+    process_table: TableState,
+    process_table_scroll_state: ScrollbarState,
+    process_table_number_of_items: usize,
+    process_details_scroll_state: ScrollbarState,
+    process_details_scroll_offset: u16,
+    process_details_number_of_lines: u16,
```
Comment (grouzen) [inline src/tui/rendering.rs:46]: I slightly rearranged and renamed these variables to make naming more uniform. I understand if you want to push these changes back to keep the old naming.

File: src/tui/rendering.rs [hunk 2]
```diff
@@ -99,16 +101,25 @@ impl Tui {
         self.search_area.insert_char(new_char);
     }
 
-    pub fn process_details_down(&mut self) {
-        self.process_details_scroll = self.process_details_scroll.saturating_add(1);
+    pub fn process_details_down(&mut self, frame: &mut Frame) {
```
Comment (grouzen) [inline src/tui/rendering.rs:104]: This method calculates the height of an area and the number of content lines dynamically so we can decide when to stop scrolling.

File: src/tui/rendering.rs [hunk 3]
```diff
@@ -127,10 +138,10 @@ impl Tui {
         self.process_table.selected()
     }
 
-    pub fn update_number_of_items(&mut self, number_of_items: usize) {
-        self.number_of_items = number_of_items;
-        self.process_table_scroll = self
-            .process_table_scroll
+    pub fn update_process_table_number_of_items(&mut self, number_of_items: usize) {
```
Comment (grouzen) [inline src/tui/rendering.rs:141]: Renamed this method according to the variable name.

File: src/tui/rendering.rs [hunk 4]
```diff
@@ -258,6 +261,9 @@ impl Tui {
     ) {
         let selected_process = search_results.nth(self.get_selected_row_index());
         let lines = process_details_lines(selected_process);
+
+        self.update_process_details_number_of_lines(area, selected_process);
```
Comment (grouzen) [inline src/tui/rendering.rs:265]: I'm unsure if updating the state variable during rendering smells, but I couldn't figure out how to do this correctly.

**Comment by jacek-kurlit** [comment]
Thank you for PR I will check it tomorrow

Cheers

**Review by jacek-kurlit — APPROVED** [review]
Thanks for he PR!

File: src/tui/rendering.rs (hunk 1 above)
Comment (jacek-kurlit) [reply src/tui/rendering.rs:46]: It's ok

File: src/tui/rendering.rs (hunk 3 above)
Comment (jacek-kurlit) [reply src/tui/rendering.rs:141]: Ok

**Comment by jacek-kurlit** [comment]
Seems like formatting failed

File: src/tui/rendering.rs (hunk 4 above)
Comment (jacek-kurlit) [reply src/tui/rendering.rs:265]: I don't have any better idea so lets go with it for now.

**Comment by grouzen** [comment]
Fixed!

**Comment by grouzen** [comment]
Oh, have you seen that `pik` is "tool of the week" on https://terminaltrove.com? I think that's cool, congrats!

**Pull request merged**